api/turing/service/ensembling_job_service_test.go

```diff
@@ -0,0 +1,201 @@
+						},
+					},
+					Ensembler: &batchensembler.Ensembler{
+						Uri: "gs://bucket-name/my-ensembler/artifacts/ensembler",
+						Result: &batchensembler.Ensembler_Result{
+							ColumnName: "prediction_score",
+							Type:       batchensembler.Ensembler_FLOAT,
+							ItemType:   batchensembler.Ensembler_FLOAT,
+						},
+					},
+					Sink: &batchensembler.Sink{
+						Type: batchensembler.Sink_BQ,
+						Columns: []string{
+							"customer_id as customerId",
+							"target_date",
+							"results",
+						},
+						SaveMode: batchensembler.Sink_OVERWRITE,
+						Config: &batchensembler.Sink_BqConfig{
+							BqConfig: &batchensembler.Sink_BigQuerySinkConfig{
+								Table:         "project.dataset.ensembling_results",
+								StagingBucket: "bucket-name",
+								Options: map[string]string{
+									"partitionField": "target_date",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func TestSaveAndFindByIDEnsemblingJobIntegration(t *testing.T) {
+	t.Run("insertion with no errors", func(t *testing.T) {
+		database.WithTestDatabase(t, func(t *testing.T, db *gorm.DB) {
+			ensemblingJobService := NewEnsemblingJobService(db)
+
+			projectID := models.ID(1)
+			ensemblerID := models.ID(1000)
+			ensemblingJob := generateEnsemblingJobFixture(1, ensemblerID, projectID)
+			err := ensemblingJobService.Save(ensemblingJob)
+			assert.NoError(t, err)
+			assert.NotEqual(t, models.ID(0), ensemblingJob.ID)
+
+			found, err := ensemblingJobService.FindByID(
+				ensemblingJob.ID,
+				EnsemblingJobFindByIDOptions{ProjectID: &projectID},
+			)
+			assert.NoError(t, err)
+
+			assert.NotEqual(t, found.ID, models.ID(0))
+			assert.Equal(t, found.Name, ensemblingJob.Name)
+			assert.Equal(t, found.EnsemblerID, ensemblingJob.EnsemblerID)
+			assert.Equal(t, found.ProjectID, ensemblingJob.ProjectID)
+			assert.Equal(t, found.EnvironmentName, ensemblingJob.EnvironmentName)
+			assert.Equal(t, models.State("pending"), ensemblingJob.Status)
+			assert.Equal(t, found.InfraConfig, ensemblingJob.InfraConfig)
+			assert.Equal(t, found.EnsemblerConfig, ensemblingJob.EnsemblerConfig)
+		})
+	})
+}
+
+func TestFindPendingJobsAndUpdateIntegration(t *testing.T) {
+	t.Run("insertion with no errors", func(t *testing.T) {
+		database.WithTestDatabase(t, func(t *testing.T, db *gorm.DB) {
+			ensemblingJobService := NewEnsemblingJobService(db)
+
+			// Save job
+			projectID := models.ID(1)
+			ensemblerID := models.ID(1000)
+			ensemblingJob := generateEnsemblingJobFixture(1, ensemblerID, projectID)
+			err := ensemblingJobService.Save(ensemblingJob)
+			assert.NoError(t, err)
+			assert.NotEqual(t, models.ID(0), ensemblingJob.ID)
+
+			// Query pending ensembling jobs
+			ensemblingJobs, err := ensemblingJobService.FindPendingJobs(10)
+			assert.NoError(t, err)
+			assert.Equal(t, 1, len(ensemblingJobs))
+
+			queriedEnsemblingJob := ensemblingJobs[0]
+			assert.Equal(t, models.State("pending"), queriedEnsemblingJob.Status)
```

Comment:
`models.State("pending")` as well as other job statuses should be replaced with a named constants

Reply:
Actually, it's already defined as `models.JobPending`

Reply:
The rationale here is that we don't want to reference the same constant that was being sent to the database. What if `models.JobPending` = `jobnotpending` instead. This test is supposed to catch this small (no reprecussion) bug as well.

Reply:
Does it matter what's the value of `models.JobPending` if in all the codebase you use the same constant to check the state of the job? `models.JobPending` can be stored as `123` or `😎😇🤗`, it doesn't matter as long as you stick to the convention, that `models.JobPending` represents jobs in pending state

Reply:
Yeah I agree with your point but let's say there was a problem in production, we take a peep into the database to see whats going on and we find that the status is `123`, it's one extra mental hoop to go through the codebase to find out what `123` means. Nonetheless, I don't have a strong enough opinion so I'll change this.

Reply:
done